Спринт 5

jarith opened the pull request.

FellonOne committed:
Инициализация проект (#1)

* add webpack;
add babel;
add typescript;
add react, react-dom and react-router-dom;
add types;
configure webpack and babel for work with typescript and react;

* update gitignore, add ".idea"

* update file structure;
update tsconfig.json;
update package.json;

* update directory structure;

* remove package-lock.json;

Co-authored-by: Nikita Ivanov <nikita@payme.uz>

art4vm committed:
Feature/20 code quality (#2)

* Минорные правки

Перенесены глобальные конфиги в корень проекта
Поправлены имена участников, версия и путь к webpack.config.js

* Качество кода (#20)

Добавлен editorconfig
Добавлены линтеры ESLint и StyleLint
Добавлен механизм вызова хуков через Husky
Настроен pre-commit hook
Добавлены команды запуска ESLint линтера
Добавлено описание работы с кодом в README

art4vm committed:
Feature/25 heroku (#3)

* Интеграция с Heroku (#25)

Добавлен expressjs для запуска локального сервера
Добавлен Procfile для настройки Heroku web процесса
Добавлен Dockerfile для возможности самостоятельного деплоя образа
 приложения
Добавлены скрипты для работы с сервером, docker и Heroku
Добавлена верификация yarn.lock в pre-commit hook
Описание README обновлено

* Update README.md

kdeit committed:
Feature/feature welcome screens (#4)

* config: gitignore + lint run

* config: package json lint run command

* fix: server error for find index.html in dist

* init scss

* feat: font init

* config: gitignore

* process: add button comp

* feat: copyright

* fix: font from base64 to file

* config: package.json dependency

* fix: file rename

* config: add eslint console rule

* fix: change color variables name

* fix: app component refactor

* config: webpack use scss only

* fix: add semantic, change welcome headers

* feat: add pages folder + fix: css

* fix: classes components to functional

* config: eslint

* config: package

* refactor: export import modules

* config: disable default export rule

art4vm committed:
Состояние приложения (#37) (#8)

Добавлены redux redux-saga и react-redux
Добавлено хранилище и действия для пользователя и авторизации
Добавлен lodash
Добавлена обработка const enum в babel

art4vm committed:
API (#38) (#7)

Добавлен axios
Добавлен класс с клиентскими запросами к API
 https://ya-praktikum.tech/api/v2
Поправлены настройки babel для работы с async await
Исправлены другие мелкие ошибки

art4vm committed:
Feature/structure (#24) (#6)

* Fix патч, для настроек качества кода

Добавлены правила для ESLint
Добавлено правило для git, для обеспечения целостности переносов строк

* Cтруктура приложения (#24)

Добавлена маршрутизация приложения
Добавлен макет аутентификации
Добавлен перехватчик запросов в API для мониторинга запросов в логах

art4vm committed:
Исправлена ошибка при обработке ответа с сервера регистрации (#12)

art4vm committed:
Добавлена saga для регистрации (#43) (#11)

art4vm committed:
Feature/52 refactoring (#13)

* Основной шрифт заменен на 'Press Start 2P' (#55)

* Рефакторинг (#52)

Рефакторинг стилей
  Исправлены названия
  Приведены к camelCase параметры sass
  Нормализованы св-ва согласно правилам stylelint
  Вынесены в параметры URL assets
  Добавлены css св-ва с цветами
  Изменены единицы размеров шрифтов с px на rem

Добавлено выполнение stylelint в pre-commit
Поправлено описание в README

art4vm committed:
Исправление работы с redux devtools (#15)

Без установленного __REDUX_DEVTOOLS приложение не запускается

art4vm committed:
Небольшие исправления настроек проекта (#19)

Изменен параметр нормализации окончаний строк при checkin в git
Теперь бинарные файлы игнорируются
Возвращены рабочие спрайты для персонажей
Исправлен регистр каталогов Button и Copyright
Исправлен каталог сборки приложения
Добавлен алиас для абсолютных путей в сборку webpack

kdeit committed:
feature/Feature-error-pages (#16)

* add page

* fix: css refactor

* fix: linter

* fix: remove unused class

* Update app/src/pages/ErrorPage/ErrorPage.tsx

Co-authored-by: art4vm <69515971+art4vm@users.noreply.github.com>

* Update app/src/pages/ErrorPage/ErrorPage.tsx

Co-authored-by: art4vm <69515971+art4vm@users.noreply.github.com>

* typo: change biggest size

Co-authored-by: art4vm <69515971+art4vm@users.noreply.github.com>

FellonOne committed:
feature/authorization-registration-forms*** (#17)

* add authorization and registration pages;
add Formik library;

* fix: css refactor

* lock dependency; rename files;

* remove formControl and FormControl;

* add FormControl directory;

* update styles;

Co-authored-by: Nikita Ivanov <nikita@payme.uz>
Co-authored-by: Dima Kuznetcov <dimakuzn@gmail.com>
Co-authored-by: art4vm <69515971+art4vm@users.noreply.github.com>

art4vm committed:
Подготовка к верстке страниц (#20)

Текущие стили UI вынесены в отдельный UiLayout
Удалены неиспользуемые и лишние стили

art4vm committed:
Feature/47 page leaderboard (#22)

* Реализация leaderboard ч.1 (#47)

Добавлен объект ApiCharacterInfo
Добавлено хранилище и sagas
Вынесены в отдельные файлы rootReducer и rootSaga

* Реализация leaderboard ч.2 (#47)

Добавлен компонент страницы
Добавлен компонент записи

art4vm committed:
Feature/51 page game (#23)

* Реализация игры ч.1 (#51)

Добавлен объект ApiCharacterInfo
Добавлено хранилище
Вынесен в отдельный файл rootReducer

* Реализация игры ч.2 (#51)

Добавлен компонент страницы
Добавлен компонент UI игры
Реализовала простейшая стаб логика движения персонажа

art4vm committed:
Изменение способа валидации поля телефона (#26)

Удалена зависимость yup-phone
Валидация поля телефона переделана на regexp
Создан утилитный файл, в который добавлен объект с regexp
 для валидации полей
Добавлен анализатор webpack-bundle-analyzer
Добавлено описание вызова анализатора в README

Дополнительно: README файл немного переформатирован

FellonOne committed:
feature/authentication-redux-connect (#21)

* add using actions on registration and authorization pages;

* update registration and authorization pages -> add connect from redux;

Co-authored-by: Nikita Ivanov <nikita@payme.uz>
Co-authored-by: art4vm <69515971+art4vm@users.noreply.github.com>

kdeit committed:
Feature/feature user edit (#24)

* page init

* merge with develop

* password change form add

* form 3 variant ready

* add tab to edit page

* use ui-layout

* link => button

* return t-center

* vhange import type

* remove unused default classes

* class name change case

* add memo to page-edit component

* remove unused imports

* login change min valid value

* Update app/src/components/UserDataChange/UserDataChange.tsx

Co-authored-by: art4vm <69515971+art4vm@users.noreply.github.com>

froziq commented on README.md:
```diff
@@ -1 +1,133 @@
+
+## Установка и запуск
+1. Клонировать репозиторий
+2. Выполнить команду `yarn` (`npm install`)
```
если проект собирается через `yarn`, лучше не мешать с `npm`, в редких случаях могут быть конфликты в yarn.lock и package-lock.json. пусть будет инструкция для одного файлового менеджера, который вы используете в проекте

froziq commented on README.md:
```diff
@@ -1 +1,133 @@
+-   Отступы `4` кроме `json` файлов, в которых `2` отступа
+-   Ширина рабочей области `120` символов
+-   В конце каждого файла кроме `md` обязательно пустая строка
+-   Окончание строк обозначается через `lf`
```
здорово, что есть гайд по основным параметрам форматирования

froziq commented on .stylelintrc.json:
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": ["stylelint-config-idiomatic-order"],
```
очень круто, что следите за такими незначительными вещами как порядок свойств, которые сильно повышают читаемость css

froziq commented on server/src/server.ts:
```diff
@@ -0,0 +1,16 @@
+});
+
+server.listen(PORT, () => {
+    // eslint-disable-next-line no-console
```
лучше убрать eslint-disable, у вас стоит правило для консолей с warn опцией, пусть в коде подсвечиваются места, которые потенциально в будущем можно порефакторить, на сборку этот ворнинг никак не повлияет

art4vm replied:
В прекоммите учитываются не только ошибки линтера но и ворнинги (ключ  --max-warnings=0)

froziq commented on app/src/store/User/sagas.ts:
```diff
@@ -0,0 +1,49 @@
+        }
+    } catch (error) {
+        // eslint-disable-next-line no-console
+        console.log('[userSignInWorker error] ', error);
```
хорошо бы показывать в интерфейсе ошибку, чтоб было понятно, выполнялся вообще запрос или нет. сейчас можно сидеть и кликать по форме с неверными данными для входа и гадать – это сайт не работает или данные для входа неверные.

froziq replied:
p.s. для вывода ошибок в консоль можно написать отдельный логгер простенький, чтоб не вызывать console.log и не писать eslint-disable в каждом месте. логгер бы выводить саму ошибку error, а так же можно выводить тело ошибки, чтоб было видно текст ошибки, который пришел с сервера

art4vm replied:
На вывод ошибок есть отдельная задача по выводу в toaster. Пока не реализовано

froziq commented on app/src/pages/RegistrationPage/RegistrationPage.tsx:
```diff
@@ -0,0 +1,96 @@
+                    <Link to="/login" className="mr-5">
+                        Authorization
+                    </Link>
+                    <Button type="submit">Register</Button>
```
после успешной регистрации хорошо бы пользователя редиректить на страницу с игрой. сейчас ничего не происходит

art4vm replied:
Это исправлено в другом патче, который еще не зашел. Редирект происходит на уровне роутера

froziq commented on app/src/router/AppRouter/AppRouter.tsx:
```diff
@@ -0,0 +1,45 @@
+                <Route path="/registration">
+                    <RegistrationPage />
+                </Route>
+                <Route path="/login">
```
предлагаю к PrivateRoute добавить ещё и UnauthorizedRoute для путей /login и /registration, чтобы уже авторизованные пользователи не могли зайти на них. потому что сейчас без вывода ошибок в интерфейс и без редиректов после авторизации сложно понять, что происходит: я зарегистрировал нового юзера; с формы никуда уйти не могу с помощью интерфейса, кнопки ведут только либо на форму авторизации, либо регистрации; захожу на форму авторизации, ввожу данные только что созданного пользователя, ничего не происходит (потому что после регистрации у нас кука сразу установилась и мы сразу аавторизуемся).

В общем без использования devTools пока тяжело разобраться что происходить и начать играть

art4vm replied:
Это как раз сделано в https://github.com/PraktikumSeoulDevTeam/castleofghosts/compare/feature/62-page-menu

froziq commented on app/src/pages/WelcomePage/WelcomePage.tsx:
```diff
@@ -0,0 +1,28 @@
+                    <div>Ghosts</div>
+                </h1>
+                <Link to="/game" className="t-center mt-5">
+                    <Button>Start</Button>
```
сейчас после нажатия на Start меня перебрасывает на роут /login, хотя я авторизован, кука у меня валидная

art4vm replied:
Это исправлено в другом патче, который еще не зашел.
Вот ветка
https://github.com/PraktikumSeoulDevTeam/castleofghosts/compare/feature/62-page-menu

froziq commented on app/src/store/Game/reducer.ts:
```diff
@@ -0,0 +1,31 @@
+import {cloneDeep} from 'lodash';
```
lodash умеет в tree shaking – импорт не всей библиотеки, а только необходимой её части, поэтому лучше импортировать вот так `import cloneDeep from 'lodash/clonedeep';`

https://lodash.com/per-method-packages
https://tyapk.ru/blog/post/correct-way-to-import-lodash-methods

froziq commented on app/src/store/index.ts:
```diff
@@ -0,0 +1,15 @@
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+// eslint-disable-next-line no-underscore-dangle
```
забавный каскад отключений правил ))

можно удалить `// eslint-disable-next-line no-underscore-dangle` потому что оно уже ни на что не влияет, ts-ignore делает всё

froziq commented on app/src/pages/EditPage/EditPage.tsx:
```diff
@@ -0,0 +1,49 @@
+import React, {useState, memo} from 'react';
```
в `AuthorizationPage.tsx` были отделены пустой строкой импорты библиотек от относительных импортов, здесь всё вместе написано, это так и должно быть ?

froziq commented on app/src/pages/RegistrationPage/RegistrationPage.tsx:
```diff
@@ -0,0 +1,96 @@
+import {connect, ConnectedProps} from 'react-redux';
+import {Link} from 'react-router-dom';
+import * as Yup from 'yup';
+import {signUpAction} from '../../store/User/actions';
```
видимо, только в одном месте были отделены импорты библиотек пустой строкой от относительный импортов, хорошо бы оформить в одном стиле все импорты по проекту, либо отделять, либо нет

art4vm replied:
Разделений пустыми строками в импортах не предусмотрено правилами, по крайней мере пока.

froziq commented on app/src/pages/ErrorPage/types.ts:
```diff
@@ -0,0 +1,5 @@
+import {RouteProps} from 'react-router-dom';
+
+export interface ErrorPageProps extends RouteProps {
+    type: '404' | '409' | '500';
```
коды ошибок лучше вынести в enum, чтоб переиспользовать и не ошибиться при написании вручную

froziq commented on app/src/pages/ErrorPage/ErrorPage.tsx:
```diff
@@ -0,0 +1,41 @@
+        case '404':
+            errorTxt = 'Ничего не найдено';
+            break;
+        case '409':
```
409 немного другой смысл передаёт и крайне редко встречается в системах. `'У вас нет доступа'` – это 403 код ошибки. 409 – это конфликт на стороне сервера, описание для него можно оставить примерно такое `'Невозможно выполнить операцию в настоящий момент'`

froziq commented on app/src/pages/ErrorPage/ErrorPage.tsx:
```diff
@@ -0,0 +1,41 @@
+            </div>
+            <div className="t-center">
+                <div>{errorTxt}</div>
+                <div className="mt-2" />
```
для чего этот div ? кажется, он лишний, класс лучше перенести на div ниже с текстом:

```javascript
<div className="t-center">
    <div>{errorTxt}</div>
    <div className="mt-2">Мы скоро всё исправим.</div>
    <div>А пока можно начать с самого начала.</div>
    <Link className="mt-5" to="/">
```

froziq commented on app/src/pages/GamePage/GamePage.tsx:
```diff
@@ -0,0 +1,27 @@
+
+    return (
+        <UiLayout className="game-page">
+            <div className="game-page__container" style={{width: width + 4, height: height + 50}}>
```
что за магические числа 4 и 50 ? почему именно такие значения ? их лучше вынести в константы и дать говорящие названия, например:

```javascript
const verticalBorder = 2;
const horizontalBorder = 25;
...
style={{width: width + verticalBorder * 2, height: height + horizontalBorder * 2}}
```

froziq commented on app/src/store/Leaderboard/reducer.ts:
```diff
@@ -0,0 +1,42 @@
+    switch (action.type) {
+        case LB_ACTION_TYPES.ADD: {
+            const newState = cloneDeep(state);
+            newState.list = [...newState.list, ...action.payload];
```
если newList уже скопировать через cloneDeep, то здесь можно проще сделать без деструктуризации `newState.list = newState.list.concat(action.payload);`

froziq commented on app/webpack.config.js:
```diff
@@ -0,0 +1,52 @@
+    resolve: {
+        extensions: ['.ts', '.tsx', '.js'],
+        alias: {
+            '~': path.resolve(__dirname, './src')
```
алиасы хорошо бы использовать для импортов в ts коде, а то сейчас все импорты через относительные пути выглядят громоздко

art4vm committed:
Исправлены недочеты в отображении форм (#29)

FellonOne committed:
fix issues with export; (#31)

run eslint lint:fix for UserDataChange component;

Co-authored-by: Nikita Ivanov <nikita@payme.uz>
Co-authored-by: art4vm <69515971+art4vm@users.noreply.github.com>

art4vm committed:
Добавлен компонент меню игры (#34)

Компонент WelcomePage был преобразован в меню.
Упрощена навигация. Теперь после авторизации достаточно перейти в
 корневой роут.
Удалены настройки для обратной навигации в PrivateRoute
В PrivateRoute добавлен параметр publicExclusive для роутов, которые
доступны только неавторизованным пользователям

kdeit committed:
add acition + saga (#35)

art4vm committed:
Внесены изменения в layout (#33)

Добавлено св-во компонента UiLayout isStatic устанавливающее высоту в
 100 vh
Внесены изменения в настройки страницы Leaderboard после правки UiLayout

art4vm commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Sprint 6](https://trello.com/c/99oGwmTO/30-sprint-6)
